substrate/frame/staking/src/lib.rs

```diff
@@ -1285,21 +1285,34 @@ impl<T: Config, const DISABLING_LIMIT_FACTOR: usize> DisablingStrategy<T>
 
 		// We don't disable more than the limit
 		if currently_disabled.len() >= Self::disable_limit(active_set.len()) {
+			log!(
+				debug,
+				"Won't disable: reached disabling limit {:?}",
+				Self::disable_limit(active_set.len())
+			);
 			return None
 		}
 
 		// We don't disable for offences in previous eras
-		if Pallet::<T>::current_era().unwrap_or_default() > slash_era {
```

Comment:
@paritytech/staking-core the naming `current_era` is super misleading and have bitten us already